pupil_src/shared_modules/surface_tracker/surface_tracker_offline.py

```diff
@@ -0,0 +1,785 @@
+                                    idx,
+                                    gp["timestamp"],
+                                    gp["norm_pos"][0],
+                                    gp["norm_pos"][1],
+                                    gp["norm_pos"][0] * surface.real_world_size["x"],
+                                    gp["norm_pos"][1] * surface.real_world_size["y"],
+                                    gp["on_surf"],
+                                    gp["confidence"],
+                                )
+                            )
+            # save fixations on surf as csv.
+            with open(
+                os.path.join(
+                    metrics_dir, "fixations_on_surface" + surface_name + ".csv"
+                ),
+                "w",
+                encoding="utf-8",
+                newline="",
+            ) as csvfile:
+                csv_writer = csv.writer(csvfile, delimiter=",")
+                csv_writer.writerow(
+                    (
+                        "start_timestamp",
+                        "norm_pos_x",
+                        "norm_pos_y",
+                        "x_scaled",
+                        "y_scaled",
+                        "on_surf",
+                    )
+                )
+                for idx, fix_on_surf in enumerate(
+                    self.fixations_on_surf_buffer[surf_idx]
+                ):
+                    idx += in_mark
+                    if fix_on_surf:
+                        without_duplicates = dict(
+                            [(fix["base_data"][1], fix) for fix in fix_on_surf]
+                        ).values()
+                        for fix in without_duplicates:
+                            csv_writer.writerow(
+                                (
+                                    self.g_pool.timestamps[idx],
+                                    idx,
+                                    fix["timestamp"],
+                                    fix["norm_pos"][0],
+                                    fix["norm_pos"][1],
+                                    fix["norm_pos"][0] * surface.real_world_size["x"],
+                                    fix["norm_pos"][1] * surface.real_world_size["y"],
+                                    fix["on_surf"],
+                                    fix["confidence"],
+                                )
+                            )
+
+            logger.info(
+                "Saved surface gaze and fixation data for '{}'".format(surface.name)
+            )
+
+            if surface.within_surface_heatmap is not None:
+                logger.info("Saved Heatmap as .png file.")
+                cv2.imwrite(
+                    os.path.join(metrics_dir, "heatmap" + surface_name + ".png"),
+                    surface.within_surface_heatmap,
+                )
+
+        logger.info("Done exporting reference surface data.")
+        # TODO enable export of surface image?
```

Comment:
Please include the code or delete it. Even if you delete it, git will remember and you can later restore. The problem with out-commented code in production code is that no one will ever delete it because it might be valuable, but slowly it will diverge from the surrounding code.